src/components/JSONExportDialog.tsx

```diff
@@ -0,0 +1,135 @@
+import React, { useState } from "react";
+import { ActionsManagerInterface } from "../actions/types";
+import { NonDeletedExcalidrawElement } from "../element/types";
+import { t } from "../i18n";
+import { useIsMobile } from "./App";
+import { AppState } from "../types";
+import { Dialog } from "./Dialog";
+import { exportToFileIcon, link, save } from "./icons";
+import { ToolButton } from "./ToolButton";
+import { actionSaveAsScene } from "../actions/actionExport";
+import { Card } from "./Card";
+
+import "./ExportDialog.scss";
+
+export const ErrorCanvasPreview = () => {
+  return (
+    <div>
+      <h3>{t("canvasError.cannotShowPreview")}</h3>
+      <p>
+        <span>{t("canvasError.canvasTooBig")}</span>
+      </p>
+      <em>({t("canvasError.canvasTooBigTip")})</em>
+    </div>
+  );
+};
+
+export type ExportCB = (
+  elements: readonly NonDeletedExcalidrawElement[],
+  scale?: number,
+) => void;
+
+const JSONExportModal = ({
+  elements,
+  appState,
+  actionManager,
+  onExportToBackend,
+}: {
+  appState: AppState;
+  elements: readonly NonDeletedExcalidrawElement[];
+  actionManager: ActionsManagerInterface;
+  onExportToBackend?: ExportCB;
+  onCloseRequest: () => void;
+}) => {
+  const exportedElements = elements;
+
+  return (
+    <div className="ExportDialog">
+      <div
+        style={{
+          display: "grid",
+          gridTemplateColumns: "repeat(auto-fit, minmax(200px, 1fr))",
+          justifyItems: "center",
```

Comment:
lets not use inline style, it makes difficult to override css in consumers (boils down to using !important)

Reply:
Moved into a class in this instance, but I wouldn't want the above be a consideration. Once we start using utility classes hosts will need to revert to hacks anyway. If we want to support customizing CSS we should  have an explicit API/workflow.